lib/modules/accountSwitcher.js

```diff
@@ -434,12 +456,42 @@ function _notifySwitchedAccountElsewhere() {
 		message,
 	});
 }
-
+function autoSwitcher(thing) {
+	if(thing.getAuthor() != loggedInUser() && window.location.search.indexOf('accountSwitcher=auto') == -1 ) {
+		switchTo(thing.getAuthor());
+		queryParameter = "auto"
+	}
+	if(window.location.search.indexOf('accountSwitcher=auto') > -1 && thing.getAuthor() == loggedInUser() && thing.isPost()) {
+		Notifications.showNotification({
+			moduleID: module.moduleID,
+			notificationID: 'auto-switched',
+			closeDelay: 4000,
+			optionKey: 'autoSwitchOnPost',
+			header: i18n('accountSwitcherAutoSwitchNotificationTitle'),
+			message: i18n('accountSwitcherAutoSwitchPostDesc', loggedInUser()),
+		});
+	}
+	if(window.location.search.indexOf('accountSwitcher=auto') > -1 && thing.getAuthor() == loggedInUser() && thing.isComment()) {
+		Notifications.showNotification({
+			moduleID: module.moduleID,
+			notificationID: 'auto-switched',
+			closeDelay: 4000,
+			optionKey: 'autoSwitchOnComment',
+			header: i18n('accountSwitcherAutoSwitchNotificationTitle'),
+			message: i18n('accountSwitcherAutoSwitchCommentDesc', loggedInUser()),
+		});
+	}
+}
 function manageAccounts() {
 	SettingsNavigation.loadSettingsPage(module.moduleID, 'accounts');
 }
 
 function reloadPage() {
 	NeverEndingReddit.resetReturnToPage();
-	location.reload();
+	if(queryParameter) {
+		window.location.href = window.location.pathname+"?"+$.param({'accountSwitcher':queryParameter})
```

Comment:
This approach is problematic for two reasons:

1. The `accountSwitcher?` could be copied into URLs which are shared and then launch a confusing pop-up to people who simply click the link.
1. This implementation, `${pathname}?accountSwitcher=foo`, doesn't preserve existing query parameters, such as context, depth, limit

A smoother approach would store "recently reloaded ${location.href} to switch accounts to ${username} " in RES storage, akin to [Comment Hide Persistor](https://github.com/honestbleeps/Reddit-Enhancement-Suite/blob/456cc098577439b906f9f21e3554515b353a8572/lib/modules/commentHidePersistor.js#L21).

Reply:
I very much agree with you, I felt icky writing that code but it was the only solution I could think of to retain data across the page refresh to login. I'm trying to learn the Storage module by reading the Comment Hide Persistor code but it's kind of confusing for someone who doesn't have much ES6 knowledge. Is there any documentation available? If this is changed this should fix some other changes you requested as well retaining to the Query Parameter.